### ISSUE TITLE
Resource Monitor marked as changed but didn't change, and fails when  re-apply  it second time 

### ISSUE THREAD

**Issue by toni-moreno** [Snowflake-Labs/terraform-provider-snowflake#1716]
**Provider & terraform Version**
```
Terraform v1.4.4
on linux_amd64
+ provider registry.terraform.io/aidanmelen/snowsql v1.3.3
+ provider registry.terraform.io/snowflake-labs/snowflake v0.61.0
```

**Describe the bug**

Terraform is marking as changed a  `snowflake_resource_monitor` resource but it didn't change.

**Expected behavior**

When re-plan without changes it marks resource as changed , when  re-apply it fails 

**Code samples and commands**

```tf
resource_monitors = {
  "RM_ETL" = {
    quota           = 554
    frequency       = "NEVER"
    start_timestamp = "2023-04-14 12:00"
    end_timestamp   = "2023-11-01 12:00"
    notify_triggers = [75, 100]
    warehouses      = ["ETL"]
  }
}

variable "resource_monitors" {
  description = "lista de resource monitors"
  type = map(object({
    for_account               = optional(bool, false)
    quota                     = number 
    frequency                 = optional(string)
    start_timestamp           = string
    end_timestamp             = string
    notify_triggers           = list(number)
    suspend_trigger           = optional(number) 
    suspend_immediate_trigger = optional(number) 
    notify_users              = optional(list(string))
    warehouses                = optional(set(string))
  }))
 #....some validations here
}


resource "snowflake_resource_monitor" "RM" {

  for_each = var.resource_monitors

  set_for_account = each.value.for_account

  name         = each.key
  credit_quota = each.value.quota

  frequency       = each.value.frequency
  start_timestamp = each.value.start_timestamp
  end_timestamp   = each.value.end_timestamp

  notify_triggers           = each.value.notify_triggers
  suspend_trigger           = each.value.suspend_trigger
  suspend_immediate_trigger = each.value.suspend_immediate_trigger

  notify_users = each.value.notify_users
  warehouses   = each.value.warehouses

  depends_on = [snowflake_warehouse.warehouse]

}

```

When executing first `terraform apply` everything seems ok  but  in snowflake ui  the time is not the same than expected form my terraform code 

![image](https://user-images.githubusercontent.com/5883405/231852580-9312d39d-3b1e-42db-8633-df3fe9730851.png)

My Snowflake account is in `switzerland-north.azure`  and my timezone is `Europe/Madrid`.

When re-plan without changes the provider is showing me the resource as changed.

```
  # snowflake_resource_monitor.RM["RM_ETL"] will be updated in-place
  ~ resource "snowflake_resource_monitor" "RM" {
      ~ end_timestamp             = "2023-11-01T05:00:00-07:00" -> "2023-11-01 12:00"
        id                        = "RM_ETL"
        name                      = "RM_ETL"
      ~ start_timestamp           = "2023-04-14T05:00:00-07:00" -> "2023-04-14 12:00"
        # (7 unchanged attributes hidden)
    }
```
When re-apply ( force it ) it fails
```
╷
│ Error: error updating resource monitor RM_ETL
│ 090259 (42601): Must specify frequency and start time together.
│ 
│   with snowflake_resource_monitor.RM["RM_ETL"],
│   on 001_resource_monitors.tf line 5, in resource "snowflake_resource_monitor" "RM":
│    5: resource "snowflake_resource_monitor" "RM" {
│ 
╵
```

Here there is 3 things to review .

1)  what timezone is taking the provider by default
2)  how can I change in the `end_timestamp` and `start_timestamp` parameters ? 
3)  ( *the bug*)  why is talking about changed  parameters that indeed didn't change
4)  (*the bug*)  why is failing when force re-apply.

**Comment by toni-moreno** [comment]
reviewing the error, seems like TIMESTAMP_OUTPUT_FORMAT is the default `YYYY-MM-DD HH24:MI:SS.FF3 TZHTZM` and when the provider runs a `SHOW RESOURCE MONITORS`  here:

https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/d9f634a1fe1911a9f2f5ad44d39b9ff5e21ff602/pkg/snowflake/resource_monitor.go#L261-L278

It gets the following timestamp 

![image](https://user-images.githubusercontent.com/5883405/231879842-f34346ef-85ee-41f6-aa10-42103da4f318.png)

that is different than used in the resource
 this is becouse terraform thinks it has changed
 ```
 # snowflake_resource_monitor.RM["RM_ETL"] will be updated in-place
  ~ resource "snowflake_resource_monitor" "RM" {
      ~ end_timestamp             = "2023-11-01T05:00:00-07:00" -> "2023-11-01 12:00"
        id                        = "RM_ETL"
        name                      = "RM_ETL"
      ~ start_timestamp           = "2023-04-14T05:00:00-07:00" -> "2023-04-14 12:00"
        # (7 unchanged attributes hidden)
    }
```

But the resource doesn't accept other format than "YYYY-MM-DD HH:MM" as input.  when trying to change timestamp to the default output format 

```tf
"RM_ETL" = {
    quota     = 100
    frequency = "NEVER"
    start_timestamp = "2023-29-03T05:00:00-07:00"
    end_timestamp   = "2023-11-01T05:00:00-07:00"
    notify_triggers = [75, 100]
    warehouses      = ["ETL"]
  }
```

this error happens.
```
Invalid date/time format string '2023-29-03T05:00:00-07:00': Invalid format: "2023-29-03T05:00:00-07:00" is malformed at "T05:00:00-07:00"
```
the solution seems to change TIMESTAMP_OUTPUT_FORMAT to the same format than supported in input  before the  `SHOW RESOURCES MONITOR` in the current terraform opened session.

`alter session set TIMESTAMP_OUTPUT_FORMAT = 'YYYY-MM-DD HH24:MI'`

 * There is any way to change the terraform session behavior to handle resource monitor ?

**Comment by sfc-gh-jcieslak** [comment]
Hello 👋
Please refer to this response - https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/1175#issuecomment-2331300488. I'll let you know whenever the updated resource is available. I think altering the session in the provider currently wouldn't make sense because we are recreating the Snowflake client a few times during the `terraform apply`, which could make altering session parameters useless. I guess the only way (at least in the current version of the provider) is to use the values proposed by the Snowflake, so they will match during the Read. In the later versions any timestamp will be possible to define.

**Comment by sfc-gh-jcieslak** [comment]
Hey 👋
The new and refactored resource monitor was released yesterday in version 0.96.0 of the provider. Please migrate with [migration guide](https://github.com/Snowflake-Labs/terraform-provider-snowflake/blob/main/MIGRATION_GUIDE.md#v0950--v0960) and let me know if the issue has been resolved so we could close the ticket. Thanks in advance 🙏.

**Comment by sfc-gh-jcieslak** [comment]
Closing due to long inactivity. Please, create another issue if you think the problem is still not resolved.
